mappyfile/pprint.py

```diff
@@ -498,6 +510,22 @@ def _format(self, composite, level=0):
             s = self.whitespace(level, 0) + type_.upper()
             lines.append(s)
 
+        length = 0
```

Comment:
Maybe move this block to its own function for clarity (and would make it easier to test edge cases)